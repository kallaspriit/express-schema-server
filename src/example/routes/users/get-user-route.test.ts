import * as supertest from 'supertest';
import setupApp from '../../app';
import {transformUser} from './users';

let app: supertest.SuperTest<supertest.Test>;

describe('get-user-route', () => {
	beforeEach(async () => {
		app = supertest(await setupApp());
	});

	it('should return registered user info', async () => {
		const createResponse = await app.post('/users').send({
			name: 'Jack Daniels',
			email: 'jack@daniels.com',
		});

		expect(createResponse.status).toEqual(200);
		expect(createResponse.body.success).toBe(true);
		expect(createResponse.body.payload).toMatchSnapshot();

		const getResponse = await app.get(`/users/${createResponse.body.payload.id}`).send();

		expect(getResponse.status).toEqual(200);
		expect(getResponse.body.success).toBe(true);
		expect(getResponse.body.payload).toMatchSnapshot();
	});

	it('should return validation error for invalid user id', async () => {
		const getResponse = await app.get('/users/abc').send();

		expect(getResponse.status).toEqual(400);
		expect(getResponse.body.success).toBe(false);
		expect(getResponse.body.validationErrors).toMatchSnapshot();
	});

	it('should return 404 not found for non-existing user', async () => {
		const getResponse = await app.get(`/users/666`).send();

		expect(getResponse.status).toEqual(404);
		expect(getResponse.text).toMatchSnapshot();
	});

	it('transformUser removes excessive data', async () => {
		const fullInfo = {
			name: 'Jack Daniels',
			email: 'jack@daniels.com',
			password: 'xxx',
		};

		const transformedInfo = transformUser(fullInfo);

		expect(transformedInfo).toMatchSnapshot();
	});
});
