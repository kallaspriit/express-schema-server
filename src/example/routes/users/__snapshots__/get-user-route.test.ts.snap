// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-user-route should return 404 not found for non-existing user 1`] = `undefined`;

exports[`get-user-route should return registered user info 1`] = `
Object {
  "email": "jack@daniels.com",
  "id": 1,
  "name": "Jack Daniels",
}
`;

exports[`get-user-route should return registered user info 2`] = `
Object {
  "email": "jack@daniels.com",
  "id": 1,
  "name": "Jack Daniels",
}
`;

exports[`get-user-route should return validation error for invalid user id 1`] = `null`;

exports[`get-user-route transformUser removes excessive data 1`] = `
Object {
  "email": "jack@daniels.com",
  "id": 0,
  "name": "Jack Daniels",
}
`;
